pkg/threadas/threadas.go

```diff
@@ -0,0 +1,92 @@
+//go:build darwin
+// +build darwin
+
+// Package threadas uses some thread trickery to facilitate
+// dropping permissions on a single thread. This allows us to call
+// underlying darwin APIs which change behavior based on the userid.
+//
+// Note that this changes permission on a single thread. It cannot
+// change permission on something like osquery (communication is over
+// a socket), and may not do the right thing if you bring in an
+// exec. This also means that if you spawn a new thread, you'll get
+// the parent process permissions.
+//
+// This is based on ideas from:
+//   * https://github.com/golang/go/issues/14592
+//   * https://wiki.freebsd.org/Per-Thread%20Credentials
+//   * https://pkg.go.dev/runtime#LockOSThread
+//
+// There's future work in supporting other platforms. Linux may have
+// `syscall(SYS_setresuid, ...)` for this.
+package threadas
+
+import (
+	"fmt"
+	"runtime"
+	"syscall"
+	"time"
+)
+
+const (
+	// KAUTH_UID_NONE and KAUTH_GID_NONE are special values. See the man page
```

Comment:
this is `man pthread_setugid_np`, correct?

Reply:
Correct